From 9e5c9e670ede0e7e401cebc7cdd121a89b23cb3c Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Mon, 19 Oct 2026 04:56:22 -0700
Subject: [PATCH] v2: remove the ZeroMQ depends package

Nothing in Lotus V2 links ZeroMQ any more. Remove the zeromq recipe,
its patch, the zmq_packages list and the NO_ZMQ switch. A NO_ZMQ=1
passed by an existing build script is now an unused make variable and
has no effect.

Tier: V2

Coordinator integration provenance (network403):
Native Qt/ZeroMQ dependency removal preserves new HTTP2/nghttp2 feature flags, helper configuration and OpenSSL3.5 chain; no removed Qt/ZMQ code is restored.
Original scripted commands/hashes above are original-author provenance; modified parent/tool/catalog hashes are not claimed to reproduce this composition. Exact coordinator adaptation scripts and ordered tree proof are in composed403-network-metrics-delivery. Build/runtime/benchmarks NOT_RUN for this composition under user compile-first policy.
---
 depends/Makefile                              |  6 ----
 depends/README.md                             |  1 -
 depends/packages/packages.mk                  |  1 -
 depends/packages/zeromq.mk                    | 34 -------------------
 .../patches/zeromq/remove_libstd_link.patch   | 25 --------------
 5 files changed, 67 deletions(-)
 delete mode 100644 depends/packages/zeromq.mk
 delete mode 100644 depends/patches/zeromq/remove_libstd_link.patch

diff --git a/depends/Makefile b/depends/Makefile
index 5034109db7787f05bd2798d29d27442c34f1e247..f4d7c27c244328a8136397f60e0e5e3a8d9fab86 100644
--- a/depends/Makefile
+++ b/depends/Makefile
@@ -38,7 +38,6 @@ NO_OPENSSL ?=
 NO_WEBTRANSPORT ?=
 NO_QR ?=
 NO_WALLET ?=
-NO_ZMQ ?=
 NO_HTTP2 ?=
 NO_JEMALLOC ?=
 BUILD_CPYTHON ?=
@@ -145,7 +144,6 @@ qt_packages_$(NO_QT) = $(qt_packages) $(qt_$(host_os)_packages) $(qt_$(host_arch
 sqlite_packages_$(NO_SQLITE) = $(sqlite_packages)
 wallet_packages_$(NO_WALLET) = $(sqlite_packages_)
 
-zmq_packages_$(NO_ZMQ) = $(zmq_packages)
 
 http2_packages_$(NO_HTTP2) = $(http2_packages)
 
@@ -171,10 +169,6 @@ ifneq ($(openssl_packages_),)
 packages += $(openssl_packages)
 endif
 
-ifneq ($(zmq_packages_),)
-packages += $(zmq_packages)
-endif
-
 ifneq ($(http2_packages_),)
 packages += $(http2_packages)
 endif
diff --git a/depends/README.md b/depends/README.md
index 2100a670873e8af1d9c99d9dbc0cd4a11f80d572..7feb5983dc5a5d89713645e9dbcbe7ce8a5070f3 100644
--- a/depends/README.md
+++ b/depends/README.md
@@ -93,7 +93,6 @@ The following can be set when running make: `make FOO=bar`
 - `CXX_STANDARD`: Set the C++ standard version used. Defaults to `c++20`.
 - `NO_QT`: Don't download/build/cache Qt and its dependencies
 - `NO_QR`: Don't download/build/cache packages needed for enabling qrencode
-- `NO_ZMQ`: Don't download/build/cache packages needed for enabling ZeroMQ
 - `NO_HTTP2`: Don't download/build/cache nghttp2, needed for HTTP/2 (`-DENABLE_HTTP2=ON`)
 - `NO_WALLET`: Don't download/build/cache libs needed to enable the wallet
 - `NO_BDB`: Don't download/build/cache BerkeleyDB
diff --git a/depends/packages/packages.mk b/depends/packages/packages.mk
index 7d73e32cf9231cb200c7864dbd0561a52d349bac..e24ffd990cb331e9a460d4f6cd419e655c6271c9 100644
--- a/depends/packages/packages.mk
+++ b/depends/packages/packages.mk
@@ -17,7 +17,6 @@ qt_mingw32_packages=qt
 
 sqlite_packages=sqlite
 
-zmq_packages=zeromq
 
 http2_packages = nghttp2
 
diff --git a/depends/packages/zeromq.mk b/depends/packages/zeromq.mk
deleted file mode 100644
index c8dda3dbf92611939462ac4f5edff3b77e3f4ef9..0000000000000000000000000000000000000000
--- a/depends/packages/zeromq.mk
+++ /dev/null
@@ -1,34 +0,0 @@
-package=zeromq
-$(package)_version=4.3.5
-$(package)_download_path=https://github.com/zeromq/libzmq/releases/download/v$($(package)_version)/
-$(package)_file_name=$(package)-$($(package)_version).tar.gz
-$(package)_sha256_hash=6653ef5910f17954861fe72332e68b03ca6e4d9c7160eb3a8de5a5a913bfab43
-$(package)_patches=remove_libstd_link.patch
-
-define $(package)_set_vars
-  $(package)_config_opts=--without-docs --disable-shared --disable-curve --disable-curve-keygen --disable-perf
-  $(package)_config_opts += --without-libsodium --without-libgssapi_krb5 --without-pgm --without-norm --without-vmci
-  $(package)_config_opts += --disable-libunwind --disable-radix-tree --without-gcov --disable-dependency-tracking
-  $(package)_config_opts += --disable-Werror --disable-drafts --enable-option-checking
-endef
-
-define $(package)_preprocess_cmds
-  patch -p1 < $($(package)_patch_dir)/remove_libstd_link.patch
-endef
-
-define $(package)_config_cmds
-  cp -f $(BASEDIR)/config.guess $(BASEDIR)/config.sub config && \
-  $($(package)_autoconf)
-endef
-
-define $(package)_build_cmds
-  $(MAKE) -j$(JOBS) src/libzmq.la
-endef
-
-define $(package)_stage_cmds
-  $(MAKE) DESTDIR=$($(package)_staging_dir) install-libLTLIBRARIES install-includeHEADERS install-pkgconfigDATA
-endef
-
-define $(package)_postprocess_cmds
-  rm -rf bin share lib/*.la
-endef
diff --git a/depends/patches/zeromq/remove_libstd_link.patch b/depends/patches/zeromq/remove_libstd_link.patch
deleted file mode 100644
index ddf91e6abfaba15442df4812af77379e7349019c..0000000000000000000000000000000000000000
--- a/depends/patches/zeromq/remove_libstd_link.patch
+++ /dev/null
@@ -1,25 +0,0 @@
-commit 47d4cd12a2c051815ddda78adebdb3923b260d8a
-Author: fanquake <fanquake@gmail.com>
-Date:   Tue Aug 18 14:45:40 2020 +0800
-
-    Remove needless linking against libstdc++
-
-    This is broken for a number of reasons, including:
-    - g++ understands "static-libstdc++ -lstdc++" to mean "link against
-      whatever libstdc++ exists, probably shared", which in itself is buggy.
-    - another stdlib (libc++ for example) may be in use
-
-    See #11981.
-
-diff --git a/src/libzmq.pc.in b/src/libzmq.pc.in
-index 233bc3a..3c2bf0d 100644
---- a/src/libzmq.pc.in
-+++ b/src/libzmq.pc.in
-@@ -7,6 +7,6 @@ Name: libzmq
- Description: 0MQ c++ library
- Version: @VERSION@
- Libs: -L${libdir} -lzmq
--Libs.private: -lstdc++ @pkg_config_libs_private@
-+Libs.private: @pkg_config_libs_private@
- Requires.private: @pkg_config_names_private@
- Cflags: -I${includedir} @pkg_config_defines@
-- 
2.50.1 (Apple Git-155)

